src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Acquisition/MissingSetupComponentRegistrationService.cs

```diff
@@ -19,8 +20,15 @@ internal class MissingSetupComponentRegistrationService : IMissingSetupComponent
     {
         private const string WasmToolsWorkloadName = "wasm-tools";
 
+        private static readonly ImmutableDictionary<string, string> s_packageVersionToComponentId = ImmutableDictionary.Create<string, string>(StringComparer.OrdinalIgnoreCase)
+            .Add("v2.1", "Microsoft.Net.Core.Component.SDK.2.1")
+            .Add("v3.1", "Microsoft.NetCore.Component.Runtime.3.1")
+            .Add("v5.0", "Microsoft.NetCore.Component.Runtime.5.0");
```

Comment:
To remove the hardcoded map, Sdk first needs to implement something on their side to pass this information using design-time.

That improvement can go in another pr.

Reply:
Osvaldo, has this been brought up with the SDK team & a backlog item been added? If so please share the work item/issue # prior to resolving this comment. We want to avoid keeping hard coded values for mapping SDK versions.

Reply:
I created this https://github.com/dotnet/project-system/issues/8186 to keep track of this.

Reply:
I suggest creating a bug in the dotnet/sdk repo about this.

One idea: They could expose this information as MSBuild items, as they do for think like the list of .NET versions supported by the SDK.